astropy/io/asdf/extension.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+import os
+
+# Make sure that all tag implementations are imported by the time we create
+# the extension class so that _astropy_asdf_types is populated correctly. We
+# could do this using __init__ files, except it causes pytest import errors in
+# the case that asdf is not installed.
+from astropy.io.asdf.tags.fits.fits import *
+from astropy.io.asdf.tags.table.table import *
+from astropy.io.asdf.tags.time.time import *
+from astropy.io.asdf.tags.transform.basic import *
+from astropy.io.asdf.tags.transform.compound import *
+from astropy.io.asdf.tags.transform.polynomial import *
+from astropy.io.asdf.tags.transform.projections import *
+from astropy.io.asdf.tags.transform.tabular import *
+from astropy.io.asdf.tags.unit.quantity import *
+from astropy.io.asdf.tags.unit.unit import *
+from astropy.io.asdf.types import _astropy_asdf_types
+
+from asdf.resolver import Resolver, DEFAULT_URL_MAPPING
+
+
+SCHEMA_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), 'schemas'))
+
+
+class AstropyExtension(object):
+    @property
+    def types(self):
+        return _astropy_asdf_types
```

Comment:
I may be missing something but it looks like `_astropy_asdf_types` includes only types with schemas defined in asdf. Are you going to create a different base type class for tags with schemas in astropy? If so, just ignore this, I will review this later.
(I am judging by code and comments in asdf/types.py.

Reply:
Yes, you are correct. If/when any schemas are provided by astropy itself, we will require a different extension for those types, along with a separate type registry. However, in the meantime it might be a good idea to name this `AstropyAsdfExtension`.